pkg/spec/spec.go

```diff
@@ -444,13 +444,24 @@ func addRlimits(config *CreateConfig, g *generate.Generator) error {
 		err error
 	)
 
+	nofileSet := false
+
 	for _, u := range config.Resources.Ulimit {
 		if ul, err = units.ParseUlimit(u); err != nil {
 			return errors.Wrapf(err, "ulimit option %q requires name=SOFT:HARD, failed to be parsed", u)
 		}
 
+		if ul.Name == "nofile" {
+			nofileSet = true
+		}
+
 		g.AddProcessRlimits("RLIMIT_"+strings.ToUpper(ul.Name), uint64(ul.Hard), uint64(ul.Soft))
 	}
+
+	if !nofileSet {
+		g.AddProcessRlimits("RLIMIT_NOFILE", 1048576, 1048576)
```

Comment:
You shold also set the RLIMIT_NOPROC to the same number
Here is the patch I did for buildah.
https://github.com/projectatomic/buildah/commit/a2b018430df1e8e89b23cb5bfa49e4d3517e1c2d